test/integration/qos_test.go

```diff
@@ -171,3 +171,112 @@ func TestQOSPlugin(t *testing.T) {
 		t.Logf("Pods were popped out in the expected order.")
 	}
 }
+
+func TestQOSPlugin_SameQoSDifferentCreationTime(t *testing.T) {
+	testCtx := &testContext{}
+	testCtx.Ctx, testCtx.CancelFn = context.WithCancel(context.Background())
+
+	cs := kubernetes.NewForConfigOrDie(globalKubeConfig)
+	testCtx.ClientSet = cs
+	testCtx.KubeConfig = globalKubeConfig
+
+	cfg, err := util.NewDefaultSchedulerComponentConfig()
+	if err != nil {
+		t.Fatal(err)
+	}
+	cfg.Profiles[0].Plugins.QueueSort = schedapi.PluginSet{
+		Enabled:  []schedapi.Plugin{{Name: qos.Name}},
+		Disabled: []schedapi.Plugin{{Name: "*"}},
+	}
+
+	testCtx = initTestSchedulerWithOptions(
+		t,
+		testCtx,
+		scheduler.WithProfiles(cfg.Profiles...),
+		scheduler.WithFrameworkOutOfTreeRegistry(fwkruntime.Registry{qos.Name: qos.New}),
+	)
+	syncInformerFactory(testCtx)
+	defer cleanupTest(t, testCtx)
+
+	ns := fmt.Sprintf("integration-test-%v", string(uuid.NewUUID()))
+	createNamespace(t, testCtx, ns)
+
+	// Create a Node.
+	nodeName := "fake-node"
+	node := st.MakeNode().Name(nodeName).Label("node", nodeName).Obj()
+	node.Status.Capacity = v1.ResourceList{
+		v1.ResourcePods:   *resource.NewQuantity(32, resource.DecimalSI),
+		v1.ResourceCPU:    *resource.NewMilliQuantity(500, resource.DecimalSI),
+		v1.ResourceMemory: *resource.NewQuantity(500, resource.DecimalSI),
+	}
```

Comment:
nit: can you add a comment to document why we use these settings? perhaps `we don't really need too many resources, so pick semi-random "low enough" values` (note this is pure guesswork of mine)